bika/lims/browser/analysisrequest/add.py

```diff
@@ -475,11 +477,12 @@ def __call__(self):
         self.context.plone_utils.addPortalMessage(message, 'info')
         # Automatic label printing won't print "register" labels for Secondary. ARs
         new_ars = [ar for ar in ARs if ar[-2:] == '01']
+        new_ar_uids = [uid_dict[ar_id] for ar_id in new_ars]
```

Comment:
Only `new_ar_uids` variable is used later, so there is no reason to set a `new_ars` and `uid_dict`variables. I suggest to declare the `new_ar_uids` before the loop (at L#459) and move the logic inside, in line L#468 as follows:
```python
    new_ar_uids = []
    for arnum, state in valid_states.items():
        ...
        ARs.append(ar.Title())
        # Automatic label printing won't print "register" label for Secondary ARs
        # TODO: Add a checkbox to Enable/Disable this in Bika Setup
        if ar.Title()[-2:] == '01':
            new_ar_uids.append(ar.UID())
```
NB: Don't understand why labels for secondary ARs shouldn't be printed automatically. I think an option in Bika Setup should be available to configure this. In any case, this is out of the scope of this functionality.